src/main/java/io/hops/hopsworks/expat/migrations/featurestore/featureview/CreateFeatureViewFromTrainingDataset.java

```diff
@@ -0,0 +1,381 @@
+    }
+  }
+
+  private void removeFeatureViewFiles(String projectName) throws RollbackException {
+    Path fvPath = getFeatureViewPath(projectName);
+    Path fvPathHdfs = new Path("hdfs://" + fvPath);
+
+    try {
+      if (!dryRun && dfso.exists(fvPath)) {
+        dfso.rm(fvPathHdfs, true);
+      }
+    } catch (IOException e) {
+      throw new RollbackException("HDFS operation failed.", e);
+    }
+  }
+
+  private void deleteFeatureView(Integer fvId) throws RollbackException {
+
+    try {
+      PreparedStatement statement = connection.prepareStatement(DELETE_FEATURE_VIEW);
+      statement.setInt(1, fvId);
+      statement.execute();
+      statement.close();
+    } catch (SQLException e) {
+      throw new RollbackException("Failed to remove feature view.", e);
+    }
+  }
+
+  private void setXAttr(Integer featurestoreId, String featureViewFullPath, String description,
+      Date createDate, String email, Integer[] features)
+      throws MigrationException {
+    try {
+      FeatureViewXAttrDTO fv = new FeatureViewXAttrDTO(featurestoreId,
+          description,
+          createDate,
+          email,
+          makeFeatures(featurestoreId, features));
+      Marshaller marshaller = jaxbContext.createMarshaller();
+      StringWriter sw = new StringWriter();
+      marshaller.marshal(fv, sw);
+      byte[] val = sw.toString().getBytes();
+      if (val.length > 13500) {
+        LOGGER.warn("xattr too large - skipping attaching features to featuregroup.");
+        marshaller.marshal(fv, sw);
+        val = sw.toString().getBytes();
+      }
+      if (!dryRun) {
+        XAttrHelper.upsertProvXAttr(dfso, featureViewFullPath,
+            FeaturestoreXAttrsConstants.FEATURESTORE, val);
+      }
+    } catch (JAXBException | XAttrException e) {
+      throw new MigrationException("Cannot set attribute.", e);
+    }
+  }
+
+  private List<FeaturegroupXAttr.SimplifiedDTO> makeFeatures(Integer featurestoreId, Integer[] featureIds)
+      throws MigrationException {
+    String getFeaturesSql = String.format(GET_FEATURES,
+        Joiner.on(" OR ").join(
+            Stream.of(featureIds).map(feature -> "`tdf`.`id` = " + feature).collect(Collectors.toList())));
+    try {
+      PreparedStatement getFeaturesStatement = connection.prepareStatement(getFeaturesSql);
+      ResultSet features = getFeaturesStatement.executeQuery();
+      Map<String, FeaturegroupXAttr.SimplifiedDTO> featureDtos = Maps.newHashMap();
+
+      while (features.next()) {
+        String featureGroupName = features.getString("fg_name");
+        Integer featureGroupVersion = features.getInt("fg_version");
+        String featureName = features.getString("name");
+        String key = featurestoreId + featureGroupName + featureGroupVersion;
+        if (featureDtos.containsKey(key)) {
+          featureDtos.get(key).getFeatures().add(featureName);
+        } else {
+          featureDtos.put(key,
+              new FeaturegroupXAttr.SimplifiedDTO(
```

Comment:
I am not sure, but there is a `FeaturegroupXattrV2`, might be that we should use this one?

Reply:
It should be `SimplifiedDTO`. https://github.com/logicalclocks/hopsworks-ee/blob/master/hopsworks-common/src/main/java/io/hops/hopsworks/common/featurestore/xattr/dto/FeatureViewXAttrDTO.java#L38